server/src/main/kotlin/com/kamelia/hedera/core/auth/SessionManager.kt

```diff
@@ -0,0 +1,79 @@
+package com.kamelia.hedera.core.auth
+
+import com.auth0.jwt.JWT
+import com.auth0.jwt.interfaces.DecodedJWT
+import com.kamelia.hedera.core.auth.store.InMemorySessionStore
+import com.kamelia.hedera.core.auth.store.SessionStore
+import com.kamelia.hedera.rest.user.UserEvents
+import com.kamelia.hedera.rest.user.UserForcefullyLoggedOutDTO
+import com.kamelia.hedera.util.Environment
+import com.kamelia.hedera.util.launchPeriodic
+import com.kamelia.hedera.util.toUUID
+import java.util.*
+import kotlin.time.Duration.Companion.milliseconds
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+
+object SessionManager {
+
+    private val store: SessionStore = InMemorySessionStore
+
+    suspend fun getUserOrNull(userId: UUID): UserState? =
+        store.getUserOrNull(userId)
+
+    suspend fun createSession(userId: UUID, userState: UserState): Session =
+        store.createSession(userId, userState)
+
+    suspend fun verify(accessToken: String): UserState? =
+        store.verify(accessToken)
+
+    suspend fun verifyRefresh(refreshToken: String): Boolean =
+        store.verifyRefresh(refreshToken)
+
+    suspend fun refreshSession(refreshToken: String): Session? {
+        val (userId, sessionId) = decodeJWT(refreshToken)
+        return store.refreshSession(userId, sessionId)
+    }
+
+    suspend fun logout(accessToken: String) {
+        val (userId, sessionId) = decodeJWT(accessToken)
+        return store.removeSession(userId, sessionId)
+    }
+
+    suspend fun logoutAllExcept(userId: UUID, sessionId: UUID, reason: String = "logout_all") {
+        UserEvents.userForcefullyLoggedOutEvent(
+            UserForcefullyLoggedOutDTO(
+                userId = userId,
+                reason = reason,
+            ),
+            ignoredSessions = listOf(sessionId)
+        )
+
+        return store.removeAllSessionsExcept(userId, sessionId)
+    }
+
+    suspend fun logoutAll(userId: UUID, reason: String = "logout_all") {
+        UserEvents.userForcefullyLoggedOutEvent(
+            UserForcefullyLoggedOutDTO(
+                userId = userId,
+                reason = reason,
+            )
+        )
+        return store.removeAllSessions(userId)
+    }
+
+    suspend fun updateSession(userState: UserState) {
+        return store.updateUserState(userState.uuid, userState)
+    }
```

Comment:
Need to send update packet through WS here

Reply:
and that's not all

![image](https://github.com/Black-Kamelia/Hedera/assets/46132319/b91e554e-96bb-4bcb-8f27-d6485dbe3388)